IPython/terminal/shortcuts.py

```diff
@@ -87,10 +87,23 @@ def create_ipython_shortcuts(shell):
 
 
 def newline_or_execute_outer(shell):
+
+    import black
     def newline_or_execute(event):
         """When the user presses return, insert a newline or execute the code."""
         b = event.current_buffer
         d = b.document
+        def try_reformat():
```

Comment:
I think it's a little bit confusing to define a function inside of another function. Could you please tell me whether there's a specific reason for this?

Reply:
For both of your questions the answer is : "there is no reason beyond pure lazyness", I was prototyping and looking at "can this be done"; you are right that a proper implementation should move the import to the top of the file and define the `try_reformat()` function outside, passing in the `b` buffer.

Reply:
Ok, fair enough :)

How do you check whether black is activated?